cmd/configschema/configschema/resolver.go

```diff
@@ -54,5 +56,62 @@ func NewDirResolver(srcRoot string, moduleName string) DirResolver {
 // PackageDir accepts a Type and returns its package dir.
 func (dr DirResolver) PackageDir(t reflect.Type) string {
 	pkg := strings.TrimPrefix(t.PkgPath(), dr.ModuleName+"/")
+	dir := localPackageDir(dr, pkg)
+	if _, err := os.ReadDir(dir); err != nil {
+		dir = externalPackageDir(dr, pkg)
+	}
+	return dir
+}
+
+// localPackageDir returns the path to a local package.
+func localPackageDir(dr DirResolver, pkg string) string {
 	return path.Join(dr.SrcRoot, pkg)
 }
+
+// externalPackageDir returns the path to an external package.
+func externalPackageDir(dr DirResolver, pkg string) string {
+	line := grepMod(path.Join(dr.SrcRoot, "go.mod"), pkg)
+	v := extractVersion(line)
+	goPath := os.Getenv("GOPATH")
+	if goPath == "" {
+		goPath = build.Default.GOPATH
+	}
+	pkgPath := buildExternalPath(goPath, pkg, v)
+	if _, err := os.ReadDir(pkgPath); err != nil {
+		panic(err)
+	}
+	return pkgPath
+}
+
+// grepMod returns the line within go.mod associated with the package
+// we are looking for.
+func grepMod(goModPath string, pkg string) string {
+	file, _ := os.ReadFile(goModPath)
+	goModFile := strings.Split(string(file), "\n")
+	var pkgPath string
+	for _, line := range goModFile {
+		if strings.Contains(line, pkg) || strings.Contains(line, DefaultModule+" ") ||
+			strings.Contains(line, "replace") && strings.Contains(line, pkg) {
```

Comment:
Wondering if we get anything useful if there's a match for DefaultModule. Also, I'm thinking `strings.Contains(line, "replace") && strings.Contains(line, pkg)` isn't necessary as written because the first `strings.Contains(line, pkg)` will already have matched, right?

Reply:
Took out the match for defaultModule and it broke the program. At some point we search for config and that case catches it. In terms of the case with the replace being short circuited, yes you are right. And with that being said I can take out the last case all together?